tests/test-dirs/destruct/issue1300.t

```diff
@@ -0,0 +1,52 @@
+From issue 1300:
+https://github.com/ocaml/merlin/issues/1300
+  $ $MERLIN single case-analysis -start 6:5 -end 6:5 -filename i1300.ml <<EOF
+  > type t =
+  > | A of int
+  > | B of int
+  > 
+  > let f = function
+  > | A x (* <<< here *)
+  > | B -> 0
+  > EOF
+  {
+    "class": "error",
+    "value": "Destruct not allowed on recovered node",
```

Comment:
This is definitely better than an assertion failure, but perhaps we could give a nicer error message?
Something like "Error: the node on which destruct was called is ill-typed"?